### DISCUSSION TITLE
OTel Demo App Bring Your Own Backend Config Files

### DISCUSSION THREAD

**Discussion by productml-ai** [open-telemetry/opentelemetry-demo#847] [Q&A] [answered]
For the OTel Demo App, I followed the guidance on [Docker deployment | OpenTelemetry](https://opentelemetry.io/docs/demo/docker-deployment/#bring-your-own-backend) to bring your own backend. Can anyone elaborate on how the OTel Collector merges the two config files of (a) otelcol-config.yml and (b) otelcol-config-extras.yml? When I add to (b), it seems to override (a) instead of it being additive. In [PR 246](https://github.com/open-telemetry/opentelemetry-demo/pull/246), there is a comment stating "...  it fully replaces the default collector config." If this is true, then do I have to add the items in (a) to ensure that the telemetry continues to get sent to Jaeger et al as well as a different vendor backend?

**Comment by julianocosta89** [comment]
@TylerHelmuth do you happen to know more about this case?
Could you help us out?

**Reply by puckpuck** [reply, marked as answer]
This would be a YAML merge. 

Objects are additive, but Arrays are overwritten.

**Reply by puckpuck** [reply]
This would be a YAML merge. 

Objects are additive, but Arrays are overwritten.

**Reply by puckpuck** [reply]
This would be a YAML merge. 

Objects are additive, but Arrays are overwritten.

**Reply by TylerHelmuth** [reply]
Details on how the collector merges config sources can be found here: https://github.com/open-telemetry/opentelemetry-collector/tree/main/confmap.  Behind the scenes it uses this library: https://github.com/knadh/koanf.  @puckpuck is correct.
